specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/ServerAzureADOnlyAuthentications.json

```diff
@@ -0,0 +1,418 @@
+          }
+        },
+        "x-ms-examples": {
+          "Gets a Azure Active Directory only authentication property.": {
+            "$ref": "./examples/AzureADOnlyAuthGet.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "ServerAzureADOnlyAuthentications"
+        ],
+        "description": "Sets Server Active Directory only authentication property or updates an existing server Active Directory only authentication property.",
+        "operationId": "ServerAzureADOnlyAuthentications_CreateOrUpdate",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "authenticationName",
+            "in": "path",
+            "description": "The name of server azure active directory only authentication.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "Default"
+            ],
+            "x-ms-enum": {
+              "name": "AuthenticationName",
+              "modelAsString": true
+            }
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The required parameters for creating or updating an Active Directory only authentication property.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ServerAzureADOnlyAuthentication"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated a Azure Active Directory only authentication property.",
+            "schema": {
+              "$ref": "#/definitions/ServerAzureADOnlyAuthentication"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ServerNotInSubscriptionResourceGroup - Specified server does not exist in the specified resource group and subscription.\n\n * 404 OperationIdNotFound - The operation with Id does not exist.\n\n * 409 OperationCancelled - The operation has been cancelled by user.\n\n * 409 OperationInterrupted - The operation on the resource could not be completed because it was interrupted by another operation on the same resource.\n\n * 500 OperationTimedOut - The operation timed out and automatically rolled back. Please retry the operation."
+          },
+          "202": {
```

Comment:
PUT should be using 201 + provisioningState.  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources-asynchronously

Reply:
@jamestao  to comment on this. All other SQL  put APIs I looked at do not send provisioning state back.. 
Customers look at operation status API which gives them provisioning state.

Reply:
As per https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#202-accepted-and-location-headers, we are using the 202 and location header pattern.